### ISSUE TITLE
softcap in logits processor

### ISSUE THREAD

**Issue by AleksKnezevic** [tenstorrent/tt-xla#4207]
Ensure softcap is being treated correctly in logits [processor](https://github.com/vllm-project/vllm/blob/0102bd2f4ce106fc2ef10e6180d882015cc81fcd/vllm/model_executor/layers/logits_processor.py#L127).

**Comment by AleksKnezevic** [comment]
@sshonTT will this be needed for the multimodal model?

**Comment by sshonTT** [comment]
@AleksKnezevic right specifically for audio input.
